[KOGITO-2562] Documention for Knative Eventing integration with Kogit… (#2941)
kiegroup/kie-docs: doc-content/kogito-docs/src/main/asciidoc/openshift/con-kogito-operator-with-knative-eventing.adoc

  resource:
    apiVersion: eventing.knative.dev/v1
    kind: Broker
    name: default
----

The `KogitoRuntime` resource used to deploy the {PRODUCT} service using the {PRODUCT} Operator must have this `KogitoInfra` instance linked to it, as shown in the following example:

.Example `KogitoRuntime` custom resource for a {PRODUCT} service with `KogitoInfra` bound to it
[source,yaml,subs="+quotes"]
----
apiVersion: app.kiegroup.org/v1alpha1
kind: KogitoRuntime
metadata:
  name: process-knative-quickstart-quarkus
spec:
  replicas: 1
  image: quay.io/__NAME_SPACE__/process-knative-quickstart-quarkus:latest
  # Reference to the KogitoInfra resource with the Knative Broker binding
  infra:
  - kogito-knative-infra
----

After you apply these resources to a Kubernetes namespace, if any BPMN or Serverless Workflow files in the {PRODUCT} project contain events to be consumed or produced, the {PRODUCT} Operator automatically creates the required Knative Eventing https://knative.dev/docs/eventing/triggers/[Triggers] for the events to be consumed and the https://knative.dev/docs/eventing/samples/sinkbinding/[`SinkBinding`] resource for the events to be produced.

> To consume events published by your application or to produce events for the Broker, you must create the required `Trigger` and `SinkBinding` resources as shown in the following examples:

Author: Hi @sterobin many thanks!

```suggestion
To consume events published by your {PRODUCT} service, you must create the required `Trigger` resource as shown in the following example:
```
The Kogito Operator would create the required resources for the Kogito service to publish and listen to messages in the Broker. The published messages can be read by any application/service/resource within the cluster, thus you create a Trigger to listen to that. We don't need to create a `SinkBinding` for our application in this scenario.

> .Example `Trigger` custom resource for consuming events

Reviewer: @ricardozanini I updated these two code snippets to match the examples we already provide in https://github.com/kiegroup/kie-docs/blob/29d0d4f05721b1a5ea70bfae82b1bacb37cc69c0/doc-content/kogito-docs/src/main/asciidoc/bpmn/proc-knative-eventing-process-services.adoc  Can you compare the snippets here and in that other file to make sure both are accurate and consistent with what we want users to do?

Author: Commented above! :)

[source,yaml]
----
apiVersion: eventing.knative.dev/v1
kind: Trigger
metadata:
  name: travelers-trigger
spec:
  # The default Broker is enabled in the cluster because the namespace is labeled with `knative-eventing-injection=enabled`.
  broker: default
  filter:
    attributes:
      # Name of the message node in the process.
      type: travelers
  # The subscriber is the deployed service. Any event that matches the filter in the Broker is sent here.
  subscriber:
    ref:
      apiVersion: v1
      kind: Service
      name: process-knative-quickstart-quarkus
----

> 

Author: The old Trigger is meant to be in here, not this one targeting the Kogito application. We need a trigger to consume messages generated by the deployed Kogito application, not a trigger for our application. :D

Too hard to understand? We might need to formulate this doc a little bit more then.

.Example `SinkBinding` custom resource for producing events
[source,yaml]
----
apiVersion: sources.knative.dev/v1alpha1
kind: SinkBinding
metadata:
  name: process-knative-quickstart-quarkus-sink
spec:
  subject:
    apiVersion: apps/v1
    kind: Deployment
    selector:
      matchLabels:
        app: process-knative-quickstart-quarkus
  # Any cloud event produced by the application is delivered to the Broker.
  sink:
    ref:
      apiVersion: eventing.knative.dev/v1
      kind: Broker
      name: default
----

> 

Author: We can remove this SinkBinding resource entirely. It's created by the operator now.
